tests/unit/framework/test_pipeline.py

```diff
@@ -125,13 +143,16 @@ def test_empty_documents_are_not_forwarded_to_other_processors(self, _):
 
     def test_not_empty_documents_are_stored_in_the_output(self, _):
         self.pipeline._setup()
+        add_empty_processor_result_to_process_mocks(self.pipeline._pipeline)
         self.pipeline._input.get_next.return_value = ({"message": "test"}, None)
         self.pipeline._store_event = mock.MagicMock()
         self.pipeline.process_pipeline()
         assert self.pipeline._store_event.call_count == 1
 
     def test_empty_documents_are_not_stored_in_the_output(self, _):
-        self.pipeline.process_event = lambda x: x.clear()
+        self.pipeline.process_event = mock.MagicMock(
+            side_effect=lambda x: (x.clear() or [ProcessorResult(name="")])
+        )
         self.pipeline._setup()
         self.pipeline._input.get_next.return_value = ({"message": "test"}, None)
         self.pipeline._store_event = mock.MagicMock()
```

Comment:
I think the assertion in the test is wrong. if we get the first message, this message should be written to the output. the second message not, because it is None. And I do not understand your side_effect